GVRf/Framework/framework/src/main/java/org/gearvrf/GVRJassimpAdapter.java

```diff
@@ -147,13 +161,22 @@ public GVRMesh createMesh(GVRContext ctx, AiMesh aiMesh, EnumSet<GVRImportSettin
         }
         if (doLighting && aiMesh.hasTangentsAndBitangents())
         {
+
             vertexDescriptor += " float3 a_tangent float3 a_bitangent";
+
             FloatBuffer tangentBuffer = aiMesh.getTangentBuffer();
-            FloatBuffer bitangentBuffer = aiMesh.getBitangentBuffer();
+
             tangentsArray = new float[tangentBuffer.capacity()];
             tangentBuffer.get(tangentsArray, 0, tangentBuffer.capacity());
-            bitangentsArray = new float[bitangentBuffer.capacity()];
-            bitangentBuffer.get(bitangentsArray, 0, bitangentBuffer.capacity());
+            bitangentsArray = new float[tangentsArray.length];
+            for(int i = 0; i < tangentsArray.length; i += 3)
+            {
+                Vector3f tangent = new Vector3f(tangentsArray[i], tangentsArray[i + 1], tangentsArray[i + 2]);
```

Comment:
Not a big deal but:

1. tangent, normal and bitagent could be allocated outside of the loop once and initialized accordingly in the loop

or

2. skip the vector3fs and directly calculate the cross product 

All that to avoid unnecessary new small object allocations. The GC is though optimized for exactly that so it is more of a style issue. Still, if avoiding the allocations is not overly complicated, why not..